src/modules/fw_pos_control_l1/fw_pos_control_l1_main.cpp

```diff
@@ -2301,7 +2305,9 @@ void FixedwingPositionControl::tecs_update_pitch_throttle(float alt_sp, float v_
 	// we're in transition
 	if (_vehicle_status.is_vtol && _vehicle_status.in_transition_mode) {
 		_was_in_transition = true;
-		_asp_after_transition = _ctrl_state.airspeed;
+		// set this to transition airspeed to init tecs correctly
+		// some vtols fly without airspeed sensor
+		_asp_after_transition = _parameters.airspeed_trans;
```

Comment:
Currently all vtol types use airspeed to transition:
tailsitter: https://github.com/PX4/Firmware/blob/master/src/modules/vtol_att_control/tailsitter.cpp#L182
tiltrotor: https://github.com/PX4/Firmware/blob/master/src/modules/vtol_att_control/tiltrotor.cpp#L211
standard: https://github.com/PX4/Firmware/blob/master/src/modules/vtol_att_control/standard.cpp#L191

The only way i see that the transition airspeed could deviate from the actual airspeed (by enough to warrant initializing tecs on that different airspeed) is when the airspeed is reached before the minimum transition time has occurred. The only other way your airspeed could deviate at this point is when you have a serious problem with your sensor in which case setting it to the transition speed seems like the safest way to handle it.

The reason i think it would be good to fix this airspeed to the transition speed is that readings are sometimes erratic and based on my initial tests it seems to be unsafe to have a wrong (too low) offset (quadchute prevented a crash again)

If you and/or @tumbili still think it's a better idea to use the reading i will make it mode dependent. A middle way could be to use the transition airspeed as a minimum value for safety reasons.

Let me know.